visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/PresentationLayer/VisitController.java

```diff
@@ -2,57 +2,52 @@
 
 
 import com.petclinic.visits.visitsservicenew.BusinessLayer.VisitService;
-import com.petclinic.visits.visitsservicenew.DataLayer.VisitDTO;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 @RestController
 @RequestMapping("visits")
+@RequiredArgsConstructor
 public class VisitController {
-
-    @Autowired
-    VisitService visitService;
+    private final VisitService visitService;
 
     @GetMapping("/{visitId}")
-    public Mono<ResponseEntity<VisitDTO>> getVisitByVisitId(@PathVariable String visitId){
+    public Mono<ResponseEntity<VisitResponseDTO>> getVisitByVisitId(@PathVariable String visitId){
         return visitService.getVisitByVisitId(visitId)
                 .map(ResponseEntity::ok)
                 .defaultIfEmpty(ResponseEntity.notFound().build());
     }
-    @GetMapping("practitioner/visits/{practitionerId}")
-    public Flux<VisitDTO> getVisitByPractitionerId(@PathVariable int practitionerId){
+    @GetMapping(value="practitioner/visits/{practitionerId}", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
+    public Flux<VisitResponseDTO> getVisitByPractitionerId(@PathVariable int practitionerId){
         return visitService.getVisitsForPractitioner(practitionerId);
     }
 
-    @PostMapping
-    public Mono<VisitDTO> addVisit(@RequestBody Mono<VisitDTO> visitDTOMono){
-        return visitService.addVisit(visitDTOMono);
+    @PostMapping("")
+    public Mono<VisitResponseDTO> addVisit(@RequestBody Mono<VisitRequestDTO> visitRequestDTOMono){
+        return visitService.addVisit(visitRequestDTOMono);
     }
 
-    @PutMapping(value = "visits/{visitId}",
-            consumes = "application/json",
-            produces = "application/json")
-    public Mono<VisitDTO> updateVisitByVisitId(@PathVariable String visitId, @RequestBody Mono<VisitDTO> visitDTOMono){
-        return visitService.updateVisit(visitId, visitDTOMono);
+    @PutMapping(value = "visits/{visitId}", consumes = "application/json", produces = "application/json")
```

Comment:
I left this value mapping as it was before but it does not follow naming conventions.